aws/resource_aws_apprunner_service.go

```diff
@@ -0,0 +1,995 @@
+						},
+						"protocol": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      apprunner.HealthCheckProtocolTcp,
+							ForceNew:     true,
+							ValidateFunc: validation.StringInSlice(apprunner.HealthCheckProtocol_Values(), false),
+						},
+						"timeout": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      2,
+							ForceNew:     true,
+							ValidateFunc: validation.IntBetween(1, 20),
+						},
+						"unhealthy_threshold": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      5,
+							ForceNew:     true,
+							ValidateFunc: validation.IntBetween(1, 20),
+						},
+					},
+				},
+			},
+
+			"instance_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cpu": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "1024",
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`1024|2048|(1|2) vCPU`), ""),
+						},
+						"instance_role_arn": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateArn,
+						},
+						"memory": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "2048",
+							ValidateFunc: validation.StringMatch(regexp.MustCompile(`2048|3072|4096|(2|3|4) GB`), ""),
+						},
+					},
+				},
+			},
+
+			"service_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"service_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"source_configuration": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"authentication_configuration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"access_role_arn": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateArn,
+									},
+									"connection_arn": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validateArn,
+									},
+								},
+							},
+						},
+						"auto_deployments_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+						"code_repository": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"code_configuration": {
+										Type:     schema.TypeList,
+										Required: true,
```

Comment:
SDK shows this as Optional
```suggestion
										Optional: true,
```